modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -69,10 +69,30 @@ module ChapelIteratorSupport {
 
   pragma "no doc"
   proc iteratorIndexType(x) type {
+   // If the result is a runtime type, we better avoid invoking
+   // ic.advance() in iteratorIndex().
```

Comment:
Yeah but we don't, for say a forall expr yielding arrays. Could we just say that any runtime component of the result is invalid?